Assets/Scripts/SS3D/Systems/Entities/Humanoid/HumanoidController.cs

```diff
@@ -53,41 +52,40 @@ public abstract class HumanoidController : NetworkActor
         public virtual float RunAnimatorValue => _runAnimatorValue;
         public bool IsRunning => _isRunning;
         #endregion
-        
-
-        protected override void OnStart()
-        {
-            base.OnStart();
 
-            Setup();
-        }
 
         protected void OnEnable()
         {
             try
             {
-                MovementControls.Enable();
-                HotkeysControls.Enable();
+                MovementControls.ToggleRun.performed += HandleToggleRun;
             }
-            catch (Exception) { }
+            catch (NullReferenceException) {}
```

Comment:
replace with` if(null && !didstart) return; else { Debug.Error(); return; }` So we can still have an error in case performed is null later in the game.